addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/HttpChannelState.java

```diff
@@ -0,0 +1,246 @@
+    );
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final ChannelUID channelUID;
+    private final ChannelTypeUID channelTypeUID;
+    private final HttpClient httpClient;
+    private final int maxHttpResponseBodyLen;
+    private final Optional<StateRequest> stateRequest;
+    private final ScheduledExecutorService scheduler;
+    private final Optional<CommandRequest> commandRequest;
+    private final BiConsumer<ChannelUID, State> stateUpdatedListener;
+    private final ErrorListener errorListener;
+
+    private volatile boolean fetchingState = false;
+    private Optional<ScheduledFuture<?>> stateUpdater = Optional.empty();
+    private Optional<String> lastStateEtag = Optional.empty();
+
+    public HttpChannelState(final ChannelUID channelUID,
+                            final ChannelTypeUID channelTypeUID,
+                            final HttpClient httpClient,
+                            final int maxHttpResponseBodyLen,
+                            final Optional<StateRequest> stateRequest,
+                            final ScheduledExecutorService scheduler,
+                            final Optional<CommandRequest> commandRequest,
+                            final BiConsumer<ChannelUID, State> stateUpdatedListener,
+                            final ErrorListener errorListener)
+    {
+        this.channelUID = channelUID;
+        this.channelTypeUID = channelTypeUID;
+        this.httpClient = httpClient;
+        this.maxHttpResponseBodyLen = maxHttpResponseBodyLen;
+        this.stateRequest = stateRequest;
+        this.scheduler = scheduler;
+        this.commandRequest = commandRequest;
+        this.stateUpdatedListener = stateUpdatedListener;
+        this.errorListener = errorListener;
+
+        this.stateRequest.ifPresent(this::startStateFetch);
+    }
+
+    /**
+     * Handles a single command.
+     *
+     * @param command the {@link Command}
+     */
+    public void handleCommand(final Command command) {
+        if (command.equals(RefreshType.REFRESH)) {
+            this.stateRequest.ifPresent(this::fetchState);
+        } else if (!this.commandRequest.isPresent()) {
+            logger.warn("[{}] Got command '{}', but no command URL set", this.channelUID.getId(), command.toFullString());
+        } else {
+            final CommandRequest commandRequest = this.commandRequest.get();
+            final HttpMethod method = commandRequest.getMethod();
+            final String commandStr = command.toFullString();
+            try {
+                final String transformedCommand = doTransform(commandRequest.getRequestTransform(), commandStr);
+                final URL transformedUrl = formatUrl(commandRequest.getUrl(), transformedCommand);
+                HttpUtil.makeRequest(
+                        this.httpClient,
+                        method.name(),
+                        transformedUrl,
+                        commandRequest.getUsername(),
```

Comment:
Could you maybe move username, password, connect timeout, request timeout into a super class ("Request") that CommandRequest and StateRequest inherit from? That way you could hand the Request object to `HttpUtil.makeRequest` to reduce this monstrosity of method signature.